### ISSUE TITLE
AVR8: Interrupt vectors should be 2 words each, not 1 word

### ISSUE THREAD

**Issue by drovak** [NationalSecurityAgency/ghidra#5013]
I'm working with some ATmega128 code. `avr8.pspec` specifies the location of each interrupt vector as 0, 1, 2..., but should be 0, 2, 4... An absolute jump instruction is four bytes long, not two. Page 61 of http://ww1.microchip.com/downloads/en/devicedoc/doc2467.pdf details the list of interrupts for this part and shows even code addresses.

**Comment by emteere** [comment]
You are correct the atmega128 does have 4 byte interrupt vectors.  I believe the original AVR8 and ATtiny have 2 byte interrupt vectors.  Newer ATtiny variants appear to be xmega architecture based and have 4 byte interrupt vectors.  I'm not an expert on the entire AVR line so I very well could be mistaken.

http://ww1.microchip.com/downloads/en/DeviceDoc/Atmel-2586-AVR-8-bit-Microcontroller-ATtiny25-ATtiny45-ATtiny85_Datasheet.pdf
https://ww1.microchip.com/downloads/en/DeviceDoc/Atmel-2543-AVR-ATtiny2313_Datasheet.pdf

If your device is an atmega, I would use the atmega256, which should be essentially compatible with that model.

We try to find the most general variants of each chip that will work in most cases for the base instruction set, memory size, registers, and addressing.

One could add all the variants of the AVR8, however many differences would only be special purpose registers or memory layout.

In the latest 10.2.x there is a script that can parse the AVR8 header files that can get the definitions for special registers for all the variants for a particular toolchain.  https://github.com/NationalSecurityAgency/ghidra/issues/2689#issuecomment-1325306884

There is a case to be made to create more separate variants with their own .pspec files that would configure all the differences.  If you are finding one of the 4 variants don't fit your architecture one could expand the basic variants.

**Comment by drovak** [comment]
Well, sure enough. I like the sound of more .pspec files for the AVR family. That seems like the appropriate direction.
